Snowflake.Data.Tests/UnitTests/SFDataConverterTest.cs

```diff
@@ -326,5 +328,25 @@ public void TestInvalidConversionInvalidDecimal(string s)
             Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(ConvertToUTF8Buffer(s), SFDataType.FIXED, typeof(decimal)));
         }
 
+        [Test]
+        [TestCase(SFDataType.TIMESTAMP_LTZ, typeof(DateTime))]
+        [TestCase(SFDataType.TIMESTAMP_TZ, typeof(DateTime))]
+        [TestCase(SFDataType.TIMESTAMP_NTZ, typeof(DateTimeOffset))]
+        [TestCase(SFDataType.TIME, typeof(DateTimeOffset))]
+        [TestCase(SFDataType.DATE, typeof(DateTimeOffset))]
+        public void TestInvalidTimestampConversion(SFDataType dataType, Type unsupportedType)
+        {
+            object unsupportedObject;
+            if (unsupportedType == typeof(DateTimeOffset))
+                unsupportedObject = new DateTimeOffset();
+            else if (unsupportedType == typeof(DateTime))
+                unsupportedObject = new DateTime();
+            else
+                unsupportedObject = null;
+
+            Assert.NotNull(unsupportedType);
+            SnowflakeDbException ex = Assert.Throws<SnowflakeDbException>(() => SFDataConverter.csharpValToSfVal(dataType, unsupportedObject));
+            SnowflakeDbExceptionAssert.HasErrorCode(ex, SFError.INVALID_DATA_CONVERSION);
+        }
```

Comment:
do we need to add anything to:
- StructuredTypesTest.TestTimeConversions
- StructuredTypesWithEmbeddedUnstructuredIT.TestSelectDateTime
- StructuredTypesWithEmbeddedUnstructuredIT.TestSelectDateTimeOffset
?

Reply:
added